bigtop-packages/src/charm/spark/layer-spark/reactive/spark.py

```diff
@@ -318,3 +318,10 @@ def client_present(client):
 def client_should_stop(client):
     if is_state('leadership.is_leader'):
         client.clear_spark_started()
+
+@when('spark.started', 'config.changed.spark_enable_thriftserver')
```

Comment:
Having read @johnsca 's doc I think this is correct. LMK @kwmonroe

Reply:
+1